modules/reactor/src/meshgenerators/AssemblyMeshGenerator.C

```diff
@@ -384,121 +420,124 @@ AssemblyMeshGenerator::AssemblyMeshGenerator(const InputParameters & parameters)
   }
   else
     declareMeshProperty("extruded", false);
+
+  // Save updates to name id map to ReactorMeshParams object
+  updateReactorMeshParams();
 }
 
 std::unique_ptr<MeshBase>
 AssemblyMeshGenerator::generate()
 {
+  // Re-initialize name_id_map and current_block_id from ReactorMeshParams object
+  // in case variables have been modified between constructor and this method
+  initializeReactorMeshParams();
```

Comment:
this is not really intuitive.
We need some sort of hasChanged() api similar to is_prepared() for the meshes.

Reply:
well it s mostly for the manual block id / name tracking so I guess it might be removed

Reply:
This has been removed now